fearless-utils/src/main/java/jp/co/soramitsu/fearless_utils/runtime/definitions/v14/TypesParserV14.kt

```diff
@@ -0,0 +1,184 @@
+import jp.co.soramitsu.fearless_utils.runtime.definitions.types.composite.DictEnum
+import jp.co.soramitsu.fearless_utils.runtime.definitions.types.composite.FixedArray
+import jp.co.soramitsu.fearless_utils.runtime.definitions.types.composite.Struct
+import jp.co.soramitsu.fearless_utils.runtime.definitions.types.composite.Tuple
+import jp.co.soramitsu.fearless_utils.runtime.definitions.types.composite.Vec
+import jp.co.soramitsu.fearless_utils.runtime.definitions.types.generics.Bytes
+import jp.co.soramitsu.fearless_utils.runtime.definitions.types.primitives.Compact
+import jp.co.soramitsu.fearless_utils.runtime.metadata.v14.LookupSchema
+import jp.co.soramitsu.fearless_utils.runtime.metadata.v14.PortableType
+import jp.co.soramitsu.fearless_utils.runtime.metadata.v14.RegistryType
+import jp.co.soramitsu.fearless_utils.runtime.metadata.v14.TypeDefArray
+import jp.co.soramitsu.fearless_utils.runtime.metadata.v14.TypeDefBitSequence
+import jp.co.soramitsu.fearless_utils.runtime.metadata.v14.TypeDefCompact
+import jp.co.soramitsu.fearless_utils.runtime.metadata.v14.TypeDefComposite
+import jp.co.soramitsu.fearless_utils.runtime.metadata.v14.TypeDefCompositeField
+import jp.co.soramitsu.fearless_utils.runtime.metadata.v14.TypeDefEnum
+import jp.co.soramitsu.fearless_utils.runtime.metadata.v14.TypeDefSequence
+import jp.co.soramitsu.fearless_utils.runtime.metadata.v14.TypeDefVariant
+import jp.co.soramitsu.fearless_utils.runtime.metadata.v14.TypeDefVariantItem
+import jp.co.soramitsu.fearless_utils.scale.EncodableStruct
+import java.math.BigInteger
+
+object TypesParserV14 {
+
+    private class Params(
+        val types: List<EncodableStruct<PortableType>>,
+        val dynamicTypeResolver: DynamicTypeResolver,
+        val typesBuilder: TypePresetBuilder
+    )
+
+    fun parse(
+        struct: EncodableStruct<LookupSchema>,
+        typePreset: TypePreset,
+        dynamicTypeResolver: DynamicTypeResolver = DynamicTypeResolver.defaultCompoundResolver()
+    ): ParseResult {
+        val builder = typePreset.newBuilder()
+        val params = Params(struct[LookupSchema.types], dynamicTypeResolver, builder)
+        parseParams(params)
+        val unknownTypes = params.typesBuilder.entries
+            .mapNotNull { (name, typeRef) -> if (!typeRef.isResolved()) name else null }
+        return ParseResult(params.typesBuilder, unknownTypes)
+    }
+
+    private fun parseParams(params: Params) {
+        for (type in params.types) {
+            val t = parseParam(params, type) ?: continue
+            params.typesBuilder.type(t)
+        }
+    }
+
+    private fun parseParam(params: Params, pt: EncodableStruct<PortableType>): Type<*>? {
+        val typesBuilder = params.typesBuilder
+        val name = pt[PortableType.id]
+        val type = pt[PortableType.type]
+        val def = type[RegistryType.def]
+        return when (def) {
+            is EncodableStruct<*> -> {
+                when {
```

Comment:
There are several cases special cases that needs to be handled:
1.  Types that needs custom encoding/decoding gets encoded as well (for example Era). To bypass it we need to have some mapping between full type path and its prebuilt implementation:
```
mapOf(
"sp_runtime::generic::era::Era" to EraType,...
)
```
)
2. For some reason some types gets encoded as struct with one field:
```
{
            "path": [
              "sp_core",
              "crypto",
              "AccountId32"
            ],
            "def": {
              "composite": {
                "fields": [
                  {
                    "type": 1,
                    "typeName": "[u8; 32]"
                  }
                ]
              }
            },
            "docs": [
              "An opaque 32-byte cryptographic identifier."
            ]
          },
``` 
To provide  better compatibility i think we need to unwrap such structures.
We can use either the way from the (1) (will require to manually check all needed structures) or we can specially handle structs with one field.